### PR TITLE
Rename retrieve_widgets to recover_lost_widgets

### PR DESCRIPTION
**Pull request by adamziel**
Trac ticket: https://core.trac.wordpress.org/ticket/53816
Trac ticket: https://core.trac.wordpress.org/ticket/53811 

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by draganescu** [comment]
I've seen the comment with "lost" in it, but I feel the `recover_lost_widgets` name is a bit too nice :) Too epic.
What are we doing, actually? It looks like a synchronization to me so maybe `sync_registered_widgets` ?

**Comment by adamziel** [comment]
@draganescu it does a lot of things and it's hard to come up with a good name – `sync_registered_widgets` works for me just as well. All I'm after here is to stop suggesting this function only reads and returns something.

**Comment by draganescu** [comment]
Thanks for renaming @adamziel. LGTM now!

**Comment by TimothyBJacobs** [comment]
This is kinda teetering on the line of [refactoring for the sake of refactoring](https://make.wordpress.org/core/handbook/contribute/code-refactoring/). If we do this rename, I think we should just soft deprecate the `retrieve_widgets` function so as not to cause unnecessary code churn that is annoying for downstream developers that support more than the latest version of WordPress to support.

**Comment by adamziel** [comment]
> This is kinda teetering on the line of refactoring for the sake of refactoring.

It has that feeling for sure. The thing is that name confused a bunch of people into thinking it's a read operation and contributed to introducing a few hard to detect bugs. Even worse, we need to call it in the GET endpoint handler – something I would like to address as one of the next steps here. For the sake of our future selves, I would rather make it obvious that it's a write.

> If we do this rename, I think we should just soft deprecate the retrieve_widgets function so as not to cause unnecessary code churn that is annoying for downstream developers that support more than the latest version of WordPress to support.

That's a very good point and something I've missed here. Let's do it!

File: src/wp-includes/widgets.php [hunk 1]
```diff
@@ -1251,13 +1251,30 @@ function _wp_sidebars_changed() {
...
  *
  * @since 2.8.0
+ * @deprecated 5.8.1 Use sync_registered_widgets()
+ * @see sync_registered_widgets()
+ *
+ * @param string|bool $theme_changed
+ * @return array
+ */
+function retrieve_widgets( $theme_changed = false ) {
+	return sync_registered_widgets( $theme_changed );
```
Comment (spacedmonkey) [inline src/wp-includes/widgets.php:1270]:
```suggestion
         _deprecated_function( __FUNCTION__, '5.8.1, 'retrieve_widgets' );
	return sync_registered_widgets( $theme_changed );
```

We should add a _deprecated_function function call here.

File: src/wp-includes/widgets.php [hunk 2]
```diff
@@ -1251,13 +1251,30 @@ function _wp_sidebars_changed() {
...
+ * Use sync_registered_widgets() instead.
  *
  * @since 2.8.0
+ * @deprecated 5.8.1 Use sync_registered_widgets()
+ * @see sync_registered_widgets()
+ *
+ * @param string|bool $theme_changed
+ * @return array
+ */
+function retrieve_widgets( $theme_changed = false ) {
```
Comment (spacedmonkey) [inline src/wp-includes/widgets.php:1269]: This function should be moved to `wp-includes/deprecated.php`

**Review by hellofromtonya — APPROVED** [review]
I agree, the new name is  better. Why? It's more clear of the reason this function exists as well as what it will do. That makes it more readable and understandable. I agree this is a good change 👍 

- All instances of `retrieve_widgets()` have been replaced ✅ 
- `retrieve_widgets()` has been soft deprecated ✅ 
- Tests `@covers` are updated ✅ 
- PHPCS passes ✅ 
- Tests pass ✅ 

This PR is ready for `commit`.

**Pull request closed without merging**